DevelopGitHub/GitHubExtension/GitHubExtension.Activity.External.Tests/TestsForExtensions/GetPagesAmountExtensionTests.cs

```diff
@@ -0,0 +1,169 @@
+    {
+        public const string HeaderForShouldNotReturnNullWhenLinkHeaderIsNotNull =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10>; rel=\"last\"";
+
+        public const string HeaderForShouldNotThrowLinkHeaderMissingException =
+           "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10>; rel=\"last\"";
+
+        public const string HeaderForShouldNotThrowLinkHeaderFormatException =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10>; rel=\"last\"";
+
+        public const string HeaderForShouldThrowLinkHeaderFormatException =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10> rel=\"last\"";
+
+        public const string HeaderForLastRelNotPresent =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\"";
+
+        public const string HeaderForShouldNotReturnNullWhenLastRelIsPresent =
+           "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=10>; rel=\"last\"";
+
+        public const string HeaderForShouldThrowLinkHeaderFormatExceptionWhenLastRelPageIsNotNumber =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=q10>; rel=\"last\"";
+
+        public const string HeaderForShouldParseLastRelHeaderCorrectly =
+            "\"<https://api.github.com/repositories/52432402/events?page=2>; rel=\"next\", <https://api.github.com/repositories/52432402/events?page=2>; rel=\"last\"";
+            
+        [Fact]
+        public void ShouldReturnNullWhenLinkHeaderIsNull()
+        {
+            // Arrange
+            IEnumerable<string> requestLinkHeader = null;
+            IGitHubEventsQuery query = Substitute.For<IGitHubEventsQuery>();
+
+            // Act
+            int? numberOfPages = query.GetNumberOfPages(requestLinkHeader);
+        
+            // Assert
+            numberOfPages.Should().NotHaveValue("link header can not be present if we are on a last page or there is only one page");
+        }
+
+        [Fact]
+        public void ShouldNotReturnNullWhenLinkHeaderIsNotNull()
+        {
+            // Arrange
+            IEnumerable<string> requestLinkHeader = new List<string>() { HeaderForShouldNotReturnNullWhenLinkHeaderIsNotNull };
+            IGitHubEventsQuery query = Substitute.For<IGitHubEventsQuery>();
+
+            // Act
+            int? numberOfPages = query.GetNumberOfPages(requestLinkHeader);
+
+            // Assert
+            numberOfPages.Should().HaveValue();
```

Comment:
It is expected to have a value, not to be null. I wrote separate test that checks exact value lower.